### PR TITLE
Update Template with new definitions, utils from JoschiZ/wcl-components

### PR DESCRIPTION
**Pull request by Ellimistdev**
re: #1  adds utils, updates definitions, fixes example typo, removes unneeded click from autotest

### CODE REVIEW

File: definitions/Template.d.ts [hunk 1]
```diff
@@ -1,43 +1,34 @@
...
+
 interface TemplateConfig {
     /**
      * Config options for the included custom plugins. Omitting a plugin or assigning a falsy value will deactivate it.
      */
     plugins: {
         clearSource?: false | ClearSourcePluginOptions
         exportString?: boolean
-        autoTest?: false | AutoTestPluginOption & {active: boolean},
-        bannerPlugin?: false | BannerPluginOptions
```
Comment (JoschiZ) [inline definitions/Template.d.ts:9]: why did you delete the BannerPluginOptions? It still exists, it's just not a custom Plugin in the Plugin folder, but included in WebPack and can be found in the webpack config.

File: definitions/Template.d.ts [hunk 2]
```diff
@@ -1,43 +1,34 @@
...
-    active: boolean,
-    options: import("webpack/types.d.ts").BannerPlugin["options"]
-}
-
 interface AutoTestPluginOption {
+    active: boolean
     loginMethod: "WCL" | "USA" | "EUROPE" | "KOREA" | "TAIWAN"
     /**This url has to lead directly to the component view (ends with &view=components)*/
-    components: Record<string, string>
+    components: {[componentName: string]: string}
```
Comment (JoschiZ) [inline definitions/Template.d.ts:35]: Is there a reason, why to prefer the {[key:type]:type} notation over using Records?

File: definitions/Template.d.ts [hunk 3]
```diff
@@ -1,43 +1,34 @@
...
 }
 
 interface ClearSourcePluginOptions {
     /**
      * If true the source code will be LZString compressed and Base64Encoded
-     * @deprecated This option will be ignored and resulting output will always be compressed
```
Comment (JoschiZ) [inline definitions/Template.d.ts]: This option is still deprecated and fully ignored in the ClearSourcePlugin.
The options object is injected but not actually used.

File: util/someImport.ts
Comment (JoschiZ) [inline util/someImport.ts:1]: Removing this breaks the example component.
The whole template should transpile out of the box and I think it should include at least one example.

File: util/anotherImport.ts
Comment (JoschiZ) [inline util/anotherImport.ts:1]: This can probably go, but someImport.ts would need to be changed accordingly.

File: definitions/Template.d.ts [hunk 4]
```diff
@@ -1,43 +1,34 @@
...
-        bannerPlugin?: false | BannerPluginOptions
+        autoTest?: false | AutoTestPluginOption
     },
     /**
      * This allows to assign individual sizes or static ids to your components.
      * It takes the component name (The file name without any endings).
      */
-    components: Record<string, Omit<Component, "component">>,
-
-    /**
```
Comment (JoschiZ) [inline definitions/Template.d.ts:17]: This is still in use in the webpack config, why did you delete ist?

**Review by JoschiZ — COMMENTED** [review]
Thank you for your PR.

I agree that some of the simpler utility like converting timestamps to readable times or maybe even the hacky "logger" could be moved over. 
However I'm not really sure if I really want to move over the more complex utility methods from my own repository into the template.
At least in their current state the code documentation is not where it would need to be to be included. I also believe that examples on their usage would also be necessary, as well as warnings about possible limitations.

I also believe you made some mistakes moving code. The project I use that contains the actual components does not use the newest version of this template.

File: definitions/Template.d.ts (hunk 2 above)
Comment (Ellimistdev) [reply definitions/Template.d.ts:35]: IIRC I was starting down a path of organizing components https://github.com/Ellimistdev/wcl-components/tree/main/components

File: definitions/Template.d.ts (hunk 3 above)
Comment (Ellimistdev) [reply definitions/Template.d.ts]: Removed erroneously.

File: definitions/Template.d.ts (hunk 1 above)
Comment (Ellimistdev) [reply definitions/Template.d.ts:9]: Couldn't say, reverted.

File: definitions/Template.d.ts (hunk 2 above)
Comment (Ellimistdev) [reply definitions/Template.d.ts:35]: That also might've been a Copilot fix to correct some typing I was working through. No strong argument either way.

**Review by JoschiZ — APPROVED** [review]
Looks good to me now.

I'm still unsure if it would be a good idea to include more, especially more complex, logic into the template. I'm open for feedback in that regard.

My initial thought would be, that that should actually live in it's own library as a npm package instead of being directly used in the template.
